src/ml_tooling/baseclass.py

```diff
@@ -139,20 +161,47 @@ def save_estimator(self, path: str) -> pathlib.Path:
                     "You haven't scored the estimator - no results available to log"
                 )
 
-            metric_scores = {self.result.metric: float(self.result.score)}
-
-            log_results(
-                metric_scores=metric_scores,
-                estimator_name=self.estimator_name,
-                estimator_params=self.result.estimator_params,
-                run_dir=self.config.RUN_DIR,
-                estimator_path=str(estimator_file),
-            )
+            log = Log.from_result(self.result, estimator_path=estimator_file)
+            log.save_log(self.config.RUN_DIR)
 
         logger.info(f"Saved estimator to {estimator_file}")
 
         return estimator_file
 
+    def dump(self):
+        if self.is_pipeline:
+            return [
+                {
+                    "name": step[0],
+                    "module": step[1].__class__.__module__,
+                    "classname": step[1].__class__.__name__,
+                    "params": step[1].get_params(),
+                }
+                for step in self.estimator.steps
+            ]
+
+        return [
+            {
+                "module": self.estimator.__class__.__module__,
+                "classname": self.estimator.__class__.__name__,
+                "params": self.estimator.get_params(),
+            }
+        ]
+
+    @classmethod
+    def load(cls, log_file: pathlib.Path):
```

Comment:
Aren't we making two load functions now? Maybe we should have .load_from_yaml and .load_from_storage

Reply:
Had changed it to `from_dict`, but that is incorrect - changed it to `from_yaml`

Reply:
As they both load the estimator - one trained and one untrained. Alternativly we should have a .load and a .specify